Dockerfile

```diff
@@ -22,3 +12,15 @@ RUN wget -O- -nv https://raw.githubusercontent.com/golangci/golangci-lint/master
 RUN wget -O - -q https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v0.9.15
 
 COPY . .
+
+RUN CGO_ENABLED=1 GOOS=linux go build -a -installsuffix cgo -o nightfall_dlp ./cmd/nightfalldlp/
+
+FROM alpine:3.8
```

Comment:
hmm how does having this 2nd image in here work?

Reply:
This is a [multi-stage docker build](https://docs.docker.com/develop/develop-images/multistage-build/#name-your-build-stages). So the first docker image is built, and inside that one we get everything required to run ci jobs and build the go binary (do all the heavy lifting). Then the second image is the image that is the final image that we would push to docker hub, which is tiny as it only contains the binary to run the action.

Using the multi-stage build allows us to generate a small final image to be run. It being small is beneficial as it can be pulled quickly from docker hub and it will take up less memory.